server.go

```diff
@@ -692,7 +743,10 @@ func (w *response) Write(m []byte) (int, error) {
 
 	switch {
 	case w.udp != nil:
-		return WriteToSessionUDP(w.udp, m, w.udpSession)
+		if u, ok := w.udp.(*net.UDPConn); ok {
+			return WriteToSessionUDP(u, m, w.udpSession)
+		}
+		return w.udp.WriteTo(m, w.pcSession)
```

Comment:
IIRC with just `Write` this should work as well? Or does it go to the default interface or some such?

Reply:
The connection is listening so we need to send back to the specific remote address. The session UDP stuff handles the specific interface, but we can’t use that for generic PacketConn’s.